R/cnd_df.R

```diff
@@ -0,0 +1,321 @@
+# ------------------------------------------------------------------------------
+# File: cnd_df.R
+# Package: sdtm.oak
+# Author: Ramiro Magno <rmagno@pattern.institute>
+# Created: 2024-05-23
+# Last Modified: 2024-05-23
+# ------------------------------------------------------------------------------
+# Description:
+#
+# This file contains functions and scripts related to the so-called conditioned
+# data frames, i.e. those data frames extended with class `cnd_df`.
+#
+# Functions:
+#
+# - `new_cnd_df()`: Create a "conditioned data set" (class cnd_df).
+# - `is_cnd_df()`: Assess whether the argument is a `cnd_df` data frame.
+# - `get_cnd_df_cnd()`: Extract the attribute "cnd" from a `cnd_df` data frame.
+# - `get_cnd_df_cnd_sum()`: Extract the attribute "cnd_sum" from a `cnd_df` data frame.
+# - `rm_cnd_df()`: De-class a cnd_df data frame.
+# - `tbl_sum.cnd_df()`: Provide a tibble header print method for `cnd_df` tibbles.
+# - `ctl_new_rowid_pillar.cnd_df()`: A print method for the row ids cnd_df` tibbles.
+# - `eval_conditions()`: Find which rows match a set of conditions.
+# - `condition_by()`: Create a conditioned data frame (user facing).
```

Comment:
Shall we change condition_by() to `condition_add`. Actually, we are adding conditions metadata to the dataframe. This also aligns with {admiral} where we have filter_add functionality.

Reply:
Sure.

Reply:
Done.